table/block_based/filter_policy.cc

```diff
@@ -48,10 +50,30 @@ Slice FinishAlwaysFalse(std::unique_ptr<const char[]>* /*buf*/) {
 class XXPH3FilterBitsBuilder : public BuiltinFilterBitsBuilder {
  public:
   explicit XXPH3FilterBitsBuilder(
-      std::atomic<int64_t>* aggregate_rounding_balance)
-      : aggregate_rounding_balance_(aggregate_rounding_balance) {}
+      std::atomic<int64_t>* aggregate_rounding_balance,
+      std::shared_ptr<CacheReservationManager> cache_res_mgr)
+      : aggregate_rounding_balance_(aggregate_rounding_balance) {
+    cache_res_mgr_ = cache_res_mgr;
+    // Reserve cache for hash_entries_ declaration: sizeof(hash_entries_)
```

Comment:
`sizeof(hash_entries_)` should be negligible (unimportant) compared to the payload of entries (which you handle below)